### PR TITLE
Dev 3451

### PR DESCRIPTION
**Pull request by charlesshale**
I've tested Orange in the pipeline using these new config options and the report output is identical.

### CODE REVIEW

File: orange/src/main/java/com/hartwig/hmftools/orange/OrangeApplication.java [hunk 1]
```diff
@@ -25,18 +26,21 @@ public class OrangeApplication {
...
+        }
 
-        OrangeConfig config = null;
         try {
-            config = OrangeConfig.createConfig(new DefaultParser().parse(options, args));
+            OrangeConfig config = OrangeConfig.createConfig(configBuilder);
+            new OrangeApplication(config).run();
         } catch (ParseException exception) {
-            LOGGER.warn(exception);
-            new HelpFormatter().printHelp(APPLICATION, options);
```
Comment (pauldwolfe) [inline orange/src/main/java/com/hartwig/hmftools/orange/OrangeApplication.java:35]: Does the help text still get printed somewhere?

File: orange/src/main/java/com/hartwig/hmftools/orange/OrangeRNAConfig.java [hunk 2]
```diff
@@ -19,93 +28,76 @@ public interface OrangeRNAConfig {
...
-        if (!hasCompleteRnaConfig(cmd,
-                RNA_SAMPLE_ID,
-                ISOFOX_GENE_DISTRIBUTION_CSV,
-                ISOFOX_ALT_SJ_COHORT_CSV,
-                ISOFOX_SUMMARY_CSV,
-                ISOFOX_GENE_DATA_CSV,
-                ISOFOX_FUSION_CSV,
-                ISOFOX_ALT_SPLICE_JUNCTION_CSV)) {
-            LOGGER.debug("No proper RNA input fed to ORANGE");
+    static OrangeRNAConfig createConfig(final ConfigBuilder configBuilder) {
```
Comment (pauldwolfe) [inline orange/src/main/java/com/hartwig/hmftools/orange/OrangeRNAConfig.java:53]: Guess this should be annotated @Nullable? Or is that not part of the current hmftools code standard?

File: orange/src/main/java/com/hartwig/hmftools/orange/OrangeRNAConfig.java [hunk 3]
```diff
@@ -19,93 +28,76 @@ public interface OrangeRNAConfig {
...
+
+        if(!anyConfigPresent) {
+            LOGGER.info("RNA config not present");
+            return null;
+        }
+
+        boolean allConfigPresent = hasRnaSampleId && hasIsofoxDir && hasGeneDistribution && hasAltSjCohortFreq;
+
+        if(!allConfigPresent) {
+            LOGGER.warn("RNA missing required config items: rnaSampleId({}) isofoxDir({}) geneCohort({}) altSjCohort({})",
```
Comment (pauldwolfe) [inline orange/src/main/java/com/hartwig/hmftools/orange/OrangeRNAConfig.java:70]: Maybe this should be an ERROR/faillure? Feels like if you've specified any of these arguments you probably intend to run RNA.

File: orange/src/main/java/com/hartwig/hmftools/orange/OrangeApplication.java (hunk 1 above)
Comment (charlesshale) [reply orange/src/main/java/com/hartwig/hmftools/orange/OrangeApplication.java:35]: Yes it does by the ConfigBuilder class

File: orange/src/main/java/com/hartwig/hmftools/orange/OrangeRNAConfig.java (hunk 2 above)
Comment (charlesshale) [reply orange/src/main/java/com/hartwig/hmftools/orange/OrangeRNAConfig.java:53]: It shouldn't be @Nullable, no - it is correct as-is.

File: orange/src/main/java/com/hartwig/hmftools/orange/OrangeRNAConfig.java (hunk 3 above)
Comment (charlesshale) [reply orange/src/main/java/com/hartwig/hmftools/orange/OrangeRNAConfig.java:70]: I kept with existing behaviour in Orange here.

**Pull request merged**